### PR TITLE
Place MechComps to turf and snap them to a neat grid

### PR DESCRIPTION
**Pull request by kbsmilk**
[GAME OBJECTS] [QOL]
## About the PR 
You can now click on the floor while holding a MechComp to put it on the floor, snapping its pixel offset to a 2x2 grid.
Also adds the generalized /obj/item/proc/place_to_turf_by_grid() and uses it for the above feature.

Parts of the new proc, such as the valid/sanity checks, are copied from /obj/proc/place_on.

## Why's this needed? 
If my mech comp monstrosities don't look pretty and neat then what's the point?

image

## Changelog 

```changelog
(u)Smilg
(*)You can now hit the floor with MechComps to place them on a neat 2x2 grid.
```

### CODE REVIEW

**Comment by LaCumbiaDelCoronavirus** [comment]
If it doesnt already, it'd be neat if you could hold a wrench in the offhand to auto wrench the components when placing them

gone will be the days of throwing components on the floor in no particular order

**Comment by klookiyam** [comment]
Can we get a picture of what this looks like?

File: code/modules/mechanics/MechanicMadness.dm [hunk 1]
```diff
@@ -494,6 +494,13 @@
 			return 1
 		return ..()
 
+	pixelaction(atom/target, params, mob/user)
+		var/turf/hit_turf = target
+		if (!istype(hit_turf) || !hit_turf || GET_DIST(hit_turf, user) > 1)
+			..()
+			return FALSE
```
[the remark this answers is not in the document]
Comment (klookiyam) [reply code/modules/mechanics/MechanicMadness.dm:501]: For point 1, you can check the density of the turf

**Comment by ZeWaka** [comment]
damn thats hot

**Review by Azrun — APPROVED** [review]

**Pull request merged**